pbj-integration-tests/src/main/proto/comparable.proto

```diff
@@ -0,0 +1,46 @@
+syntax = "proto3";
+
+package proto;
+
+option java_package = "com.hedera.pbj.test.proto.java";
+option java_multiple_files = true;
+// <<<pbj.java_package = "com.hedera.pbj.test.proto.pbj">>> This comment is special code for setting PBJ Compiler java package
+
+import "timestampTest.proto";
+import "google/protobuf/wrappers.proto";
+import "everything.proto";
+
+message ComparableTest {
+  int32 int32Number = 1;
+  double doubleNumber = 2;
+  bool booleanField = 3;
+  string text = 4;
+  bytes bytesField = 5;
+  ComparableEnum comparableEnum = 6;
+  ComparableSubObj subObject = 7;
+}
+
+enum ComparableEnum {
+  UNSPECIFIED = 0;
+  ONE = 1;
+  TWO = 2;
+  THREE = 3;
+}
+
+message ComparableSubObj {
+  int32 num = 1;
+}
+
+/**
+ * Example of a message with a limited set of fields that are comparable.
+ */
+// <<<pbj.comparable = "int32Number, text, subObject" >>>
```

Comment:
This should lead to a failure in build time, since `ComparableSubObj` is not comparable (as noted in many comments above, messages should be comparable only if explicitly annotated with pbj.comparable)

Reply:
`ComparableSubObj` is comparable. However, I added a negative test case for what you're talking about - see `non_compilable_comparable_sub_obj.proto`